packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -77,7 +83,65 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
     gridProps['aria-colcount'] = state.collection.columnCount;
   }
 
+  // Many screen readers do not announce when items in a grid are selected/deselected.
+  // We do this using an ARIA live region.
+  let selection = state.selectionManager.rawSelection;
+  let lastSelection = useRef(selection);
+  useUpdateEffect(() => {
+    if (!state.selectionManager.isFocused) {
+      return;
+    }
+
+    let addedKeys = diffSelection(selection, lastSelection.current);
+    let removedKeys = diffSelection(lastSelection.current, selection);
+
+    // If adding or removing a single row from the selection, announce the name of that item.
+    let messages = [];
+    if (addedKeys.size === 1 && removedKeys.size === 0) {
+      let addedText = getRowText(addedKeys.keys().next().value);
+      if (addedText) {
+        messages.push(formatMessage('selectedItem', {item: addedText}));
+      }
+    } else if (removedKeys.size === 1 && addedKeys.size === 0) {
+      let removedText = getRowText(removedKeys.keys().next().value);
+      if (removedText) {
+        messages.push(formatMessage('deselectedItem', {item: removedText}));
+      }
+    }
+
+    // Announce how many items are selected, except when selecting the first item.
+    if (state.selectionManager.selectionMode === 'multiple') {
+      if (messages.length === 0 || selection === 'all' || selection.size > 1 || lastSelection.current === 'all' || lastSelection.current.size > 1) {
+        messages.push(selection === 'all'
+          ? formatMessage('selectedAll')
+          : formatMessage('selectedCount', {count: selection.size})
+        );
+      }
+    }
+
+    if (messages.length > 0) {
+      announce(messages.join(' '));
```

Comment:
With VoiceOver at least, I think focus management within the table grid is causing focus on the current item to take precedence over the announcement of the selected items. Could we add `, {assertiveness: 'polite'}` to this?

Reply:
Really? In both Safari and Chrome with VO I'm getting the announcement. I tried polite but it was just not announcing at all sometimes...

Reply:
You know what? I think I accidentally conflated this with the ActionBar PR (#1934) and was testing selection announcement on one of those builds as well. Just tested using [View the storybook-17](https://reactspectrum.blob.core.windows.net/reactspectrum/48aece638cb7aea4c9dbc183da1ff1682839d146/storybook-17/index.html) and the announcement is clear as can be. Sorry.